### PR TITLE
Netlify configuration

### LINKED ISSUE #315: Switch hosting provider
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

There is already an open issue that suggest the change to Netlify #84, But I think any provider that has deployment previews and redirections should be fine.

Any good personal experience and recommendation with a hosting provider that meets the motivation of the issue will be very valuable! 

### Motivation

PR previews (#153 )
Redirections and rewrites (#312 )

### PR DESCRIPTION
**Pull request by luisorbaiceta**
This PR is intended to config Netlify as our hosting provider.

- [x] Setup redirects for old documentation
- [ ] Add our current ci pipeline to deploy process
- [ ] Add `fastify.io` domain

Closes #315 #153 #84

#### Checklist

- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: npm ci
+
+      - name: Test
+        run: npm test
+
+      # Sets the branch name as environment variable
+      - uses: nelonoel/branch-name@v1.0.1
+      - uses: jsmrcaga/action-netlify-deploy@master
+        with:
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
```
Comment (luisorbaiceta) [inline .github/workflows/ci.yml:35]: This two environment variables need to be added to the repository. @mcollina

Comment (luisorbaiceta) [reply .github/workflows/ci.yml:35]: @mcollina please let me know when this is ready 🙏

**Comment by luisorbaiceta** [comment]
I've temporarily disabled deploy previews to find a workaround for netlify github action not letting us use the generated deploy preview for lighthouse testing. If nothing comes up, I will remove lighthouse testing and leave the workflow as is right now

**Comment by Eomm** [comment]
Closing for https://github.com/fastify/website-next

**Pull request closed without merging**